### PR TITLE
Add pyGSTi recipe

### PR DESCRIPTION
**Pull request by basnijholt**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/pygsti/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  home: http://www.pygsti.info
+  summary: A python implementation of Gate Set Tomography
+  license: MIT AND Apache-2.0
+  license_file:
+    - LICENSE
+
+extra:
+  recipe-maintainers:
+    - basnijholt
+    - enielse
```
Comment (basnijholt) [inline recipes/pygsti/meta.yaml:54]: @enielse could you verify you want to be a recipe maintainer? If no, I will remove you again ;)

**Comment by basnijholt** [comment]
@conda-forge/help-python, @conda-forge/staged-recipes please review  🙌

File: recipes/pygsti/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipes/pygsti/meta.yaml:54]: Did we get a response here?

Comment (basnijholt) [reply recipes/pygsti/meta.yaml:54]: @ocefpaf, he verified this offline (a private message on Slack).

Comment (ocefpaf) [reply recipes/pygsti/meta.yaml:54]: We need it here. Sorry.

Comment (enielse) [reply recipes/pygsti/meta.yaml:54]: I'm happy to be a maintainer, thanks!

**Pull request merged**